sdk/src/azure/core/az_base64.c

```diff
@@ -0,0 +1,521 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/az_base64.h>
+#include <azure/core/internal/az_precondition_internal.h>
+
+#include <azure/core/_az_cfg.h>
+
+// The maximum integer length of binary data that can be encoded into base 64 text and still fit
+// into an az_span which has an int32_t length, i.e. (INT32_MAX / 4) * 3;
+#define _az_MAX_SAFE_ENCODED_LENGTH 1610612733
+
+#define _az_ENCODING_PAD '='
```

Comment:
Replace `#define` with unnamed enum?

Reply:
I saw this pattern being used in IoT already, so decided to follow it. We can change it across the board later if there is a coding guideline :)